hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/ExponentialRetryPolicy.java

```diff
@@ -128,6 +138,8 @@ public boolean shouldRetry(final int retryCount, final int statusCode) {
     return retryCount < this.retryCount
         && (statusCode == -1
         || statusCode == HttpURLConnection.HTTP_CLIENT_TIMEOUT
+        || statusCode == HttpURLConnection.HTTP_GONE
+        || statusCode == HTTP_TOO_MANY_REQUESTS
```

Comment:
Should we keep this in AzureADAuthentication condition.

Reason being, now if any API in AbfsHttpOperation get 429 or 410, it will be retried 30 times.
Right now what would happen in 429 / 410:
`executeHttpOperation` would give true and in completeExecute after very first call:
```
 if (result.getStatusCode() >= HttpURLConnection.HTTP_BAD_REQUEST) {
      throw new AbfsRestOperationException(result.getStatusCode(), result.getStorageErrorCode(),
          result.getStorageErrorMessage(), null, result);
    }

```
would throw exception.

Reply:
Removing from ExponentialRetryPolicy, we can have https://github.com/apache/hadoop/pull/5273/files#diff-dff9c93d1668203c206aa1c092aef9d2921dc6e20af8888d06fae34778991531R320-R321 as

```
!succeeded && isRecoverableFailure

&& (tokenFetchRetryPolicy.shouldRetry(retryCount, httperror)||httpError==429 ||httpError==410);
```

reason being, this check is only required in ADAuthenticator.

Reply:
We plan to retry for these status codes, if they come as a response from backend as well. Hence adding these into a centralized exponential retry policy class.